src/hooks/useOutsideClick.ts

```diff
@@ -0,0 +1,14 @@
+import { RefObject, useEffect } from 'react';
+
+export const useOutsideClick = (ref: RefObject<HTMLElement>, callback: () => void) => {
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      const { target } = event;
+      if (ref.current && !ref.current.contains(target as Node)) {
+        callback();
+      }
+    };
+    document.addEventListener('click', handleClickOutside);
+    return () => document.removeEventListener('click', handleClickOutside);
+  }, [ref, callback]);
```

Comment:
전체적으로 문단 사이에 여백 한줄씩 구분해주시는게 좋을거같은데 도원님 생각은 어떠신가요?

Reply:
addEventListner 사이에는 여백이 있는 게 좋을 것 같은데, 나머지는 잘 모르겠네요.
개인적으로는 선언문 다음에 공백을 넣는 게 좋을 것 같긴 한데, 선언문이 하나인 경우에는 별로인 것 같아서...
```suggestion
  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      const { target } = event;
      if (ref.current && !ref.current.contains(target as Node)) {
        callback();
      }
    };
    
    document.addEventListener('click', handleClickOutside);
    return () => document.removeEventListener('click', handleClickOutside);
  }, [ref, callback]);
```